kubectl-plugin/pkg/cmd/get/get_nodes.go

```diff
@@ -0,0 +1,267 @@
+package get
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type GetNodesOptions struct {
+	configFlags   *genericclioptions.ConfigFlags
+	ioStreams     *genericclioptions.IOStreams
+	namespace     string
+	cluster       string
+	node          string
+	allNamespaces bool
+}
+
+type node struct {
+	creationTimestamp v1.Time
+	cpus              resource.Quantity
+	gpus              resource.Quantity
+	tpus              resource.Quantity
+	memory            resource.Quantity
+	namespace         string
+	cluster           string
+	_type             string
+	workerGroup       string
+	name              string
+}
+
+var getNodesExample = templates.Examples(`
+		# Get nodes in the default namespace
+		kubectl ray get node
+
+		# Get nodes in all namespaces
+		kubectl ray get node --all-namespaces
+
+		# Get all nodes in a namespace
+		kubectl ray get node --namespace my-namespace
+
+		# Get all nodes for Ray clusters named my-raycluster in all namespaces
+		kubectl ray get node --ray-cluster my-raycluster --all-namespaces
+
+		# Get all nodes in a namespace for a Ray cluster
+		kubectl ray get node --namespace my-namespace --ray-cluster my-raycluster
+
+		# Get one node in a namespace for a Ray cluster
+		kubectl ray get node my-node --namespace my-namespace --ray-cluster my-raycluster
+	`)
+
+func NewGetNodesOptions(streams genericclioptions.IOStreams) *GetNodesOptions {
+	return &GetNodesOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+		ioStreams:   &streams,
+	}
+}
+
+func NewGetNodesCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewGetNodesOptions(streams)
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "node [NODE] [(-c|--ray-cluster) RAYCLUSTER]",
+		Aliases:      []string{"nodes"},
+		Short:        "Get Ray nodes",
+		Example:      getNodesExample,
+		SilenceUsage: true,
+		Args:         cobra.MaximumNArgs(1),
+		// ValidArgsFunction: TODO
```

Comment:
Should I add shell completion function here or leave as TODO for later PR? Leaning towards latter to not make this change too large.

Reply:
I'll create an issue to track.